### ISSUE TITLE
Contributing instructions not working in a vanilla environment

### ISSUE THREAD

**Issue by startakovsky** [joerick/pyinstrument#120]
Hello,

I thought I'd sit down this evening and knock out #119 but I was stopped when the Contributing instructions didn't work for me. After 20 minutes of troubleshooting I figured maybe the community here may have some quick insight to get me up and running. 

I'm running with pyenv, created and activated an environment with Python 3.9.0 and I'm using node v15.5.0.

Here is the output of `python setup.py build` running in my local directory:

```

(pyinstrument) ➜  pyinstrument git:(master)  python setup.py build
running build
running build_py
7.3.0
Running command: npm install

added 1482 packages, removed 7 packages, and audited 1657 packages in 23s

17 vulnerabilities (10 low, 1 moderate, 6 high)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
Running command: npm run build

> html_js@0.1.0 build
> vue-cli-service build

node:internal/modules/cjs/loader:928
  throw err;
  ^

Error: Cannot find module 'json-parse-even-better-errors'
Require stack:
- /Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/read-pkg/node_modules/parse-json/index.js
- /Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/read-pkg/index.js
- /Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/@vue/cli-service/lib/Service.js
- /Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/@vue/cli-service/bin/vue-cli-service.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:925:15)
    at Function.Module._load (node:internal/modules/cjs/loader:769:27)
    at Module.require (node:internal/modules/cjs/loader:997:19)
    at require (node:internal/modules/cjs/helpers:92:18)
    at Object.<anonymous> (/Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/read-pkg/node_modules/parse-json/index.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Module.require (node:internal/modules/cjs/loader:997:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/read-pkg/node_modules/parse-json/index.js',
    '/Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/read-pkg/index.js',
    '/Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/@vue/cli-service/lib/Service.js',
    '/Users/steven.tartakovsky/code/pyinstrument/html_renderer/node_modules/@vue/cli-service/bin/vue-cli-service.js'
  ]
}
npm ERR! code 1
npm ERR! path /Users/steven.tartakovsky/code/pyinstrument/html_renderer
npm ERR! command failed
npm ERR! command sh -c vue-cli-service build

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/steven.tartakovsky/.npm/_logs/2021-02-14T02_01_54_525Z-debug.log
Traceback (most recent call last):
  File "/Users/steven.tartakovsky/code/pyinstrument/setup.py", line 70, in <module>
    setup(
  File "/Users/steven.tartakovsky/.pyenv/versions/pyinstrument/lib/python3.9/site-packages/setuptools/__init__.py", line 165, in setup
    return distutils.core.setup(**attrs)
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/Users/steven.tartakovsky/code/pyinstrument/setup.py", line 38, in run
    self.check_call(['npm', 'run', 'build'], cwd=HTML_RENDERER_DIR)
  File "/Users/steven.tartakovsky/code/pyinstrument/setup.py", line 14, in check_call
    subprocess.check_call(args, **popen_kwargs)
  File "/Users/steven.tartakovsky/.pyenv/versions/3.9.0/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 1.
```

**Comment by startakovsky** [comment]
And, the log file referenced above is output here: 

```
0 verbose cli [
0 verbose cli   '/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/bin/node',
0 verbose cli   '/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/bin/npm',
0 verbose cli   'run',
0 verbose cli   'build'
0 verbose cli ]
1 info using npm@7.3.0
2 info using node@v15.5.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/lib/node_modules/npm/npmrc Completed in 0ms
5 timing config:load:builtin Completed in 0ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/Users/steven.tartakovsky/code/pyinstrument/html_renderer/.npmrc Completed in 0ms
9 timing config:load:project Completed in 0ms
10 timing config:load:file:/Users/steven.tartakovsky/.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 6ms
19 verbose npm-session 5f7756a8212bfa3a
20 timing npm:load Completed in 13ms
21 timing command:run-script Completed in 682ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (node:events:376:20)
22 verbose stack     at maybeClose (node:internal/child_process:1063:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:295:5)
23 verbose pkgid html_js@0.1.0
24 verbose cwd /Users/steven.tartakovsky/code/pyinstrument/html_renderer
25 verbose Darwin 20.3.0
26 verbose argv "/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/bin/node" "/Users/steven.tartakovsky/.nvm/versions/node/v15.5.0/bin/npm" "run" "build"
27 verbose node v15.5.0
28 verbose npm  v7.3.0
29 error code 1
30 error path /Users/steven.tartakovsky/code/pyinstrument/html_renderer
31 error command failed
32 error command sh -c vue-cli-service build
33 verbose exit 1
```

**Comment by joerick** [comment]
Hmm, that's odd. I just tried a clean checkout and it's working for me. What are your versions of Node and `npm` that you have installed? I have:

```
joerick@joerick4 ~/D/pyinstrument-master> node --version
v10.15.3
joerick@joerick4 ~/D/pyinstrument-master> npm --version
6.9.0
```

**Comment by joerick** [comment]
Sorry, I just saw you have node v15.5.0.... hmm

**Comment by joerick** [comment]
I've tried again with v15.5.0 and it's still working for me... not sure what's going on there to be honest. 

If you wanted to skip over that step (you'd not be editing the JS anyway) you could download a tarball from PyPI and replace `pyinstrument/renderers/html_resources/app.js` with the same file from the tarball.

But actually, I just had an idea. Could you try the following steps?

```
cd html_renderer
npm ci
npm run build
```

...and let me know the output?

**Comment by startakovsky** [comment]
Hi @joerick thanks for your reply...   those last instructions make sense because it's just supposed to follow the package-lock. I tried them and the output was unsuccessful, but different and promising. Pasting screenshot below

![image](https://user-images.githubusercontent.com/689723/107889082-57779700-6ec5-11eb-8855-e021ab1e9a80.png)

**Comment by startakovsky** [comment]
But.... when I did `nvm use 12.20.0` (as it was already installed on my machine that's why I chose it).... it seems successful. (Screenshot here)

![Uploading image.png…]()

**Comment by joerick** [comment]
Nice! Yes I should change setup.py to use 'npm ci' then

**Comment by startakovsky** [comment]
Will do so as my first PR thanks.

**Comment by startakovsky** [comment]
Actually things were erroring out again after trying python setup.py build as well as not being able to locate app.js..... so I think the best is to just change the readme to list the latest supported version of node for this. Thoughts?

**Comment by startakovsky** [comment]
Yes. I am confirming that the installation instructions are great for Contributing when I use `12.20.0`

**Comment by joerick** [comment]
Cool. Yes adding that to the readme would be great.

**Comment by joerick** [comment]
Resolved in #121
